### ISSUE TITLE
XmlUnit 2.6.3 is not compatible with AssertJ-Core 3.13.2

### ISSUE THREAD

**Issue by kaklakariada** [xmlunit/xmlunit#166]
The AssertJ dependency for XmlUnit 2.6.1 is on 2.9.0. Latest version is 3.13.2. Using assertj-core 3.13.2 causes an `AbstractMethodError`:

```
java.lang.AbstractMethodError: org.xmlunit.assertj.NodeAssertFactory.createAssert(Ljava/lang/Object;)Lorg/assertj/core/api/Assert;
	at org.assertj.core.api.FactoryBasedNavigableIterableAssert.toAssert(FactoryBasedNavigableIterableAssert.java:39)
	at org.assertj.core.api.AbstractIterableAssert.first(AbstractIterableAssert.java:2358)
```

Simply upgrading assertj-core in xmlunit to 3.13.2 breaks Java 7 compatibility: https://travis-ci.org/kaklakariada/xmlunit/jobs/601656704

Maybe it's possible to solve this similar to #135?
Thank you for your help!

**Comment by krystiankaluzny** [comment]
Hello @kaklakariada 

I notice that in AssertJ 3.13.0 `AssertFactory` interface was changed from 
```java
public interface AssertFactory<T, ASSERT>
```
to
```java
public interface AssertFactory<T, ASSERT extends Assert<?, ?>>
```
The problem is that there is no replacement like it was in #135. So if we want to keep all xmlunit-assertj features, we need to use suitable version of AssertFactory.
To achieve that I added runtime AssertFactory subclass generation.

The bug fix will be in the next XmlUnit release.
